feat: creates file/directory in active path

EstebanBorai opened the pull request:
<!--- IMPORTANT: If this PR addresses multiple unrelated issues, it will be closed until separated. -->

### Description

Uses `EditorManager` to retrieve active directory and create Files/Directories in the
current active directory.

### Related Issues

I found the todos in the source code while reading.
I wanted to address the related issue but didn't found one.

Here is the TODO comment: https://github.com/CodeEditApp/CodeEdit/blob/main/CodeEdit/Features/NavigatorSidebar/ProjectNavigator/ProjectNavigatorToolbarBottom.swift#L59

* #ISSUE_NUMBER

N/A

### Checklist

<!--- Add things that are not yet implemented above -->

- [x] I read and understood the [contributing guide](https://github.com/CodeEditApp/CodeEdit/blob/main/CONTRIBUTING.md) as well as the [code of conduct](https://github.com/CodeEditApp/CodeEdit/blob/main/CODE_OF_CONDUCT.md)
- [-] The issues this PR addresses are related to each other
- [ ] My changes generate no new warnings
- [ ] My code builds and runs on my machine
- [ ] My changes are all related to the related issue above
- [ ] I documented my code

### Screenshots


https://github.com/CodeEditApp/CodeEdit/assets/34756077/4d69b39f-b618-4528-a56b-e8699194211b

matthijseikelenboom commented:
The TODO comment is in the ProjectNavigator. You're accessing the TabBar to get the required data. While I believe that this does work, but it's not the correct way. We shouldn't want to mix those domains when not necessary. The ProjectNavigator should have a current selected file as well, so that should be accessed. As the ProjectNavigatorBottomBar is part of thé ProjectNavigator.

thecoolwinter commented:
Addressing the comment left in Discord: This is very close, instead of using the workspace document, use the editor manager directly via the following in `ProjectNavigatorToolbarBottom`:
```swift
@EnvironmentObject var editorManager: EditorManager
```

You can use the editor manager to get the active editor pane, and then the selected tab from that. See the [`EditorManager.activeTabGroup`](https://github.com/CodeEditApp/CodeEdit/blob/4a8f0938dc161a4ecdc613e9715c289138f29234/CodeEdit/Features/Editor/Models/EditorManager.swift#L20), and [`Editor.selectedTab`](https://github.com/CodeEditApp/CodeEdit/blob/4a8f0938dc161a4ecdc613e9715c289138f29234/CodeEdit/Features/Editor/Models/Editor.swift#L56) properties respectively.

That should get rid of the need to add methods to `WorkspaceDocument` and addresses the domain concern @matthijseikelenboom had.

EstebanBorai commented:
> Addressing the comment left in Discord: This is very close, instead of using the workspace document, use the editor manager directly via the following in `ProjectNavigatorToolbarBottom`:
> 
> ```swift
> 
> @EnvironmentObject var editorManager: EditorManager
> 
> ```
> 
> 
> 
> You can use the editor manager to get the active editor pane, and then the selected tab from that. See the [`EditorManager.activeTabGroup`](https://github.com/CodeEditApp/CodeEdit/blob/4a8f0938dc161a4ecdc613e9715c289138f29234/CodeEdit/Features/Editor/Models/EditorManager.swift#L20), and [`Editor.selectedTab`](https://github.com/CodeEditApp/CodeEdit/blob/4a8f0938dc161a4ecdc613e9715c289138f29234/CodeEdit/Features/Editor/Models/Editor.swift#L56) properties respectively.
> 
> 
> 
> That should get rid of the need to add methods to `WorkspaceDocument` and addresses the domain concern @matthijseikelenboom had.

Great! Thanks so much!

EstebanBorai commented:
Hi @thecoolwinter I just applied suggestions! Also added a recording for demo. Repo in demo is public available here: https://github.com/whizzes/playa

thecoolwinter reviewed (dismissed):
Awesome! Looks good to me

austincondiff commented:
@EstebanBorai Could we highlight the text after the untitled file/folder is created?

matthijseikelenboom commented on CodeEdit/Features/NavigatorArea/ProjectNavigator/ProjectNavigatorToolbarBottom.swift:
```diff
@@ -50,21 +52,56 @@ struct ProjectNavigatorToolbarBottom: View {
+                    workspace.addFile(fileName: "untitled")
+                }
+
+                // TODO: Render inactive path warning or perhaps Finder to create file?
```
This won't necessary if `activeTablURL` doesn't return an optional

EstebanBorai replied:
Gotcha! Fixed it here: https://github.com/CodeEditApp/CodeEdit/pull/1416/commits/0c3d06980ca8420235f7440bb8de23e6d87b22bd

thecoolwinter commented on CodeEdit/Features/NavigatorArea/ProjectNavigator/ProjectNavigatorToolbarBottom.swift:
```diff
@@ -50,21 +52,56 @@ struct ProjectNavigatorToolbarBottom: View {
 
+    /// Retrieves the active tab URL from the underlying editor instance, if theres no
+    /// active tab, fallbacks to the workspace's root directory
+    private func activeTabURL() -> URL? {
```
This is a perfectly fine place to return an optional. There's a couple different failure states that could happen in this method (no file manager, url components fail, no selected tab, invalid root url). Not having optionals is not a code style we try to maintain.

austincondiff reviewed (approved):
LGTM!

thecoolwinter commented:
@matthijseikelenboom can we get a review here

austincondiff commented:
@allcontributors add @EstebanBorai for code
